Issue with scheduled cleanup

```
2025-09-16T20:00:00.001Z  INFO 1 --- [CivicSage] [   scheduling-1] d.u.p.c.b.cleanup.ScheduledFileDeletion  : Looking for unused files uploaded before 2025-09-16T19:00:00.000893098Z

2025-09-16T20:00:00.012Z ERROR 1 --- [CivicSage] [   scheduling-1] o.s.s.s.TaskUtils$LoggingErrorHandler    : Unexpected error occurred in scheduled task

org.springframework.orm.jpa.JpaSystemException: A problem occurred in the SQL executor : JDBC parameter value not bound - org.hibernate.sql.exec.internal.SqlTypedMappingJdbcParameter@2afa5579

	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:348) ~[spring-orm-6.2.10.jar!/:6.2.10]

	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:256) ~[spring-orm-6.2.10.jar!/:6.2.10]

	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:241) ~[spring-orm-6.2.10.jar!/:6.2.10]

	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:560) ~[spring-orm-6.2.10.jar!/:6.2.10]

	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-6.2.10.jar!/:6.2.10]

	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:343) ~[spring-tx-6.2.10.jar!/:6.2.10]

	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:160) ~[spring-tx-6.2.10.jar!/:6.2.10]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.2.10.jar!/:6.2.10]

	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:136) ~[spring-data-jpa-3.5.3.jar!/:3.5.3]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.2.10.jar!/:6.2.10]

	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:223) ~[spring-aop-6.2.10.jar!/:6.2.10]

	at jdk.proxy2/jdk.proxy2.$Proxy170.getFileSourcesByUploadDateBeforeAndTemporaryIsTrueAndUsedByChatsIsEmpty(Unknown Source) ~[na:na]

	at de.uol.pgdoener.civicsage.business.cleanup.ScheduledFileDeletion.deleteOldFiles(ScheduledFileDeletion.java:34) ~[!/:0.0.1-SNAPSHOT]

	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]

	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]

	at org.springframework.scheduling.support.ScheduledMethodRunnable.runInternal(ScheduledMethodRunnable.java:130) ~[spring-context-6.2.10.jar!/:6.2.10]

	at org.springframework.scheduling.support.ScheduledMethodRunnable.lambda$run$2(ScheduledMethodRunnable.java:124) ~[spring-context-6.2.10.jar!/:6.2.10]

	at io.micrometer.observation.Observation.observe(Observation.java:498) ~[micrometer-observation-1.15.3.jar!/:1.15.3]

	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:124) ~[spring-context-6.2.10.jar!/:6.2.10]

	at org.springframework.scheduling.config.Task$OutcomeTrackingRunnable.run(Task.java:85) ~[spring-context-6.2.10.jar!/:6.2.10]

	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-6.2.10.jar!/:6.2.10]

	at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:96) ~[spring-context-6.2.10.jar!/:6.2.10]

	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[na:na]

	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]

	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]

	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]

	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]

	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]

Caused by: org.hibernate.sql.exec.ExecutionException: A problem occurred in the SQL executor : JDBC parameter value not bound - org.hibernate.sql.exec.internal.SqlTypedMappingJdbcParameter@2afa5579

	at org.hibernate.sql.exec.internal.AbstractJdbcParameter.bindParameterValue(AbstractJdbcParameter.java:80) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.bindParameters(DeferredResultSetAccess.java:208) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.executeQuery(DeferredResultSetAccess.java:238) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.getResultSet(DeferredResultSetAccess.java:172) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.results.jdbc.internal.JdbcValuesResultSetImpl.<init>(JdbcValuesResultSetImpl.java:74) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.resolveJdbcValuesSource(JdbcSelectExecutorStandardImpl.java:355) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.doExecuteQuery(JdbcSelectExecutorStandardImpl.java:137) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.executeQuery(JdbcSelectExecutorStandardImpl.java:102) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.exec.spi.JdbcSelectExecutor.executeQuery(JdbcSelectExecutor.java:91) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.sql.exec.spi.JdbcSelectExecutor.list(JdbcSelectExecutor.java:165) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.lambda$new$1(ConcreteSqmSelectQueryPlan.java:149) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.withCacheableSqmInterpretation(ConcreteSqmSelectQueryPlan.java:439) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.performList(ConcreteSqmSelectQueryPlan.java:359) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.query.sqm.internal.QuerySqmImpl.doList(QuerySqmImpl.java:380) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.query.spi.AbstractSelectionQuery.list(AbstractSelectionQuery.java:143) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at org.hibernate.query.Query.getResultList(Query.java:120) ~[hibernate-core-6.6.26.Final.jar!/:6.6.26.Final]

	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]

	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]

	at org.springframework.orm.jpa.SharedEntityManagerCreator$DeferredQueryInvocationHandler.invoke(SharedEntityManagerCreator.java:419) ~[spring-orm-6.2.10.jar!/:6.2.10]

	at jdk.proxy2/jdk.proxy2.$Proxy178.getResultList(Unknown Source) ~[na:na]

	at org.springframework.data.jpa.repository.query.JpaQueryExecution$CollectionExecution.doExecute(JpaQueryExecution.java:130) ~[spring-data-jpa-3.5.3.jar!/:3.5.3]

	at org.springframework.data.jpa.repository.query.JpaQueryExecution.execute(JpaQueryExecution.java:93) ~[spring-data-jpa-3.5.3.jar!/:3.5.3]

	at org.springframework.data.jpa.repository.query.AbstractJpaQuery.doExecute(AbstractJpaQuery.java:160) ~[spring-data-jpa-3.5.3.jar!/:3.5.3]

	at org.springframework.data.jpa.repository.query.AbstractJpaQuery.execute(AbstractJpaQuery.java:148) ~[spring-data-jpa-3.5.3.jar!/:3.5.3]

	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.doInvoke(RepositoryMethodInvoker.java:170) ~[spring-data-commons-3.5.3.jar!/:3.5.3]

	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.invoke(RepositoryMethodInvoker.java:158) ~[spring-data-commons-3.5.3.jar!/:3.5.3]

	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.doInvoke(QueryExecutorMethodInterceptor.java:170) ~[spring-data-commons-3.5.3.jar!/:3.5.3]

	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.invoke(QueryExecutorMethodInterceptor.java:149) ~[spring-data-commons-3.5.3.jar!/:3.5.3]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.2.10.jar!/:6.2.10]

	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:380) ~[spring-tx-6.2.10.jar!/:6.2.10]

	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-6.2.10.jar!/:6.2.10]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.2.10.jar!/:6.2.10]

	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:138) ~[spring-tx-6.2.10.jar!/:6.2.10]

	... 21 common frames omitted
```

--- a/src/main/java/de/uol/pgdoener/civicsage/business/completion/ChatRepository.java
+++ b/src/main/java/de/uol/pgdoener/civicsage/business/completion/ChatRepository.java
@@ -1,5 +1,6 @@
 package de.uol.pgdoener.civicsage.business.completion;
 
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 
@@ -11,6 +12,7 @@ import java.util.UUID;
 public interface ChatRepository
         extends CrudRepository<Chat, UUID> {
 
+    @Query("SELECT c FROM Chat c WHERE c.lastInteraction < :threshold")
     Collection<Chat> getChatsByLastInteractionBefore(OffsetDateTime threshold);
 
 }

--- a/src/main/java/de/uol/pgdoener/civicsage/business/source/FileSourceRepository.java
+++ b/src/main/java/de/uol/pgdoener/civicsage/business/source/FileSourceRepository.java
@@ -1,5 +1,6 @@
 package de.uol.pgdoener.civicsage.business.source;
 
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 
@@ -25,6 +26,7 @@ public interface FileSourceRepository
 
     Optional<FileSource> getFileSourceByHash(String hash);
 
+    @Query("SELECT f FROM FileSource f WHERE f.uploadDate < :threshold AND f.temporary = true AND f.usedByChats is empty")
     Collection<FileSource> getFileSourcesByUploadDateBeforeAndTemporaryIsTrueAndUsedByChatsIsEmpty(OffsetDateTime threshold);
 
 }

